keras/applications/imagenet_utils.py

```diff
@@ -35,26 +36,21 @@ def preprocess_input(x, data_format=None, mode='caffe'):
         x *= 2.
         return x
 
+    vgg_mean = [103.939, 116.779, 123.68]
     if data_format == 'channels_first':
         if x.ndim == 3:
             # 'RGB'->'BGR'
             x = x[::-1, ...]
```

Comment:
So the above works with symbolic tensors?

Reply:
I think yes, if my understanding of "working with symbolic tensors" is correct. (I'm new to Keras.)

```sh
(keras) ITs-MacBook-Pro:keras yangzhang$ git br
* master
  vec_vgg_preprocess_input
(keras) ITs-MacBook-Pro:keras yangzhang$ python
Python 3.6.1 |Anaconda 4.4.0 (x86_64)| (default, May 11 2017, 13:04:09) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from keras.layers import Input, Lambda
Using Theano backend.
>>> from keras.applications import vgg16, xception
>>> inputs = Input(shape=(299, 299, 3))
>>> x1 = Lambda(xception.preprocess_input, output_shape=(299, 299, 3))(inputs)
>>> x2 = Lambda(vgg16.preprocess_input, output_shape=(299, 299, 3))(inputs)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/yangzhang/git/keras/keras/engine/topology.py", line 603, in __call__
    output = self.call(inputs, **kwargs)
  File "/Users/yangzhang/git/keras/keras/layers/core.py", line 651, in call
    return self.function(inputs, **arguments)
  File "/Users/yangzhang/git/keras/keras/applications/imagenet_utils.py", line 55, in preprocess_input
    x[..., 0] -= 103.939
TypeError: 'TensorVariable' object does not support item assignment
>>> 
(keras) ITs-MacBook-Pro:keras yangzhang$ git br
* master
  vec_vgg_preprocess_input
(keras) ITs-MacBook-Pro:keras yangzhang$ git co vec_vgg_preprocess_input
Switched to branch 'vec_vgg_preprocess_input'
Your branch is up-to-date with 'origin/vec_vgg_preprocess_input'.
(keras) ITs-MacBook-Pro:keras yangzhang$ python
Python 3.6.1 |Anaconda 4.4.0 (x86_64)| (default, May 11 2017, 13:04:09) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from keras.layers import Input, Lambda
Using Theano backend.
>>> from keras.applications import vgg16, xception
>>> inputs = Input(shape=(224, 224, 3))
>>> x1 = Lambda(xception.preprocess_input, output_shape=(299, 299, 3))(inputs)
>>> x2 = Lambda(vgg16.preprocess_input, output_shape=(299, 299, 3))(inputs)
```